recorder_test.go

```diff
@@ -3,185 +3,277 @@ package lightstep
+					var value fakeFloat64 = 42.0
+					keyValue := fakeRecorder.convertToKeyValue(keyString, value)
+					Expect(keyValue.GetDoubleValue()).To(Equal(float64(42.0)))
+				})
+			})
+		})
 
-	checkCapSize(defaultMaxSpans, defaultMaxSpans)
+		Context("Of log translation", func() {
+			BeforeEach(func() {
+				fakeRecorder = Recorder{
+					maxLogKeyLen:   20,
+					maxLogValueLen: 40,
+				}
+				timestamp = time.Unix(arbitraryTimestampSecs, 0)
+				otLogs := make([]ot.LogRecord, 8)
+				for i := 0; i < 8; i++ {
+					otLogs[i] = ot.LogRecord{
+						Timestamp: timestamp,
+						Fields: []log.Field{
+							log.String("string", fmt.Sprintf("foo%d", i)),
+							log.Object("object", []interface{}{i, i, true, "suhhh"}),
+							log.String("too_long"+strings.Repeat("-", 50), strings.Repeat("-", 110)),
+						},
+					}
+				}
+				logs = fakeRecorder.translateLogs(otLogs, nil)
+			})
 
-	maxBuffer := 10
-	recorder = NewTracer(Options{
-		AccessToken:      "0987654321",
-		MaxBufferedSpans: maxBuffer,
-		UseGRPC:          true,
-	}).(basictracer.Tracer).Options().Recorder.(*Recorder)
+			Context("When a timestamp is translated", func() {
+				It("is translated with google_protobuf", func() {
+					for i := 0; i < 8; i++ {
+						Expect(logs[i].Timestamp).To(Equal(&google_protobuf.Timestamp{arbitraryTimestampSecs, 0}))
+					}
+				})
+			})
 
-	checkCapSize(0, maxBuffer)
+			Context("When a short string field is translated", func() {
+				It("is translated to String Key Value pair", func() {
+					for i := 0; i < 8; i++ {
+						Expect(logs[i].Keyvalues[0]).To(Equal(
+							&cpb.KeyValue{
+								Key:   "string",
+								Value: &cpb.KeyValue_StringValue{fmt.Sprintf("foo%d", i)},
+							},
+						))
+					}
+				})
+			})
 
-	spans = append(spans, makeSpanSlice(100*defaultMaxSpans)...)
-	for _, span := range spans {
-		recorder.RecordSpan(span)
-	}
+			Context("When an object is translated", func() {
+				It("Is marshled into JSON", func() {
+					for i := 0; i < 8; i++ {
+						pl, _ := json.Marshal([]interface{}{i, i, true, "suhhh"})
+						Expect(logs[i].Keyvalues[1]).To(Equal(
+							&cpb.KeyValue{Key: "object", Value: &cpb.KeyValue_JsonValue{string(pl)}},
+						))
+					}
+				})
+			})
 
-	checkCapSize(maxBuffer, maxBuffer)
+			Context("When a long string field is translated", func() {
+				It("Is translated to a truncated String Key Value pair", func() {
+					for i := 0; i < 8; i++ {
+						Expect(logs[i].Keyvalues[2]).To(Equal(
+							&cpb.KeyValue{
+								Key:   "too_long-----------…",
+								Value: &cpb.KeyValue_StringValue{"---------------------------------------…"},
+							},
+						))
+					}
+				})
+			})
+		})
+	})
 
-	_ = NewTracer(Options{
-		AccessToken: "0987654321",
-		UseThrift:   true,
-	}).(basictracer.Tracer).Options().Recorder.(*thrift_rpc.Recorder)
-}
+	Context("When calling Close() twice", func() {
+		Context("With grpc enabled", func() {
+			It("Should not fail", func() {
+				tracer = NewTracer(Options{
```

Comment:
Should a BeforeEach block be used for setup even when it only services one It block?